middleware/src/handlers/handlers.go

```diff
@@ -140,20 +155,31 @@ func (handlers *Handlers) versionHandler(w http.ResponseWriter, r *http.Request)
 func (handlers *Handlers) wsHandler(w http.ResponseWriter, r *http.Request) {
 	ws, err := handlers.upgrader.Upgrade(w, r, nil)
 	if err != nil {
-		log.Println(err.Error() + " Failed to upgrade connection")
+		log.Printf("Failed to upgrade connection: %s", err)
+		return
 	}
 
 	err = handlers.noiseConfig.InitializeNoise(ws)
 	if err != nil {
-		log.Println(err.Error() + "Noise connection failed to initialize")
+		log.Printf("Noise connection failed to initialize: %s", err)
 		return
 	}
+
 	server := rpcserver.NewRPCServer(handlers.middleware)
 
 	handlers.mu.Lock()
 	handlers.clientsMap[handlers.nClients] = server.RPCConnection.WriteChan()
 	handlers.runWebsocket(ws, server.RPCConnection.ReadChan(), server.RPCConnection.WriteChan(), handlers.nClients)
 	handlers.nClients++
 	handlers.mu.Unlock()
+
 	go server.Serve()
+
+	handlers.mu.Lock()
+	for _, event := range handlers.eventQueue {
+		for k := range handlers.clientsMap {
+			handlers.clientsMap[k] <- event.Identifier
+		}
+	}
+	handlers.mu.Unlock()
```

Comment:
My understanding of go channels is still quite poor. I hope this works out nicely. Only fear I have here is that the channel will need to hold events until the Server is up and listening.

Reply:
I'm not sure if I understand what you mean. Can you be a bit more specify?

Reply:
It's probably fine, but the rpc server might not be running yet even after a connected client reacts to a notification with a rpc call.